guacamole-common-js/src/main/webapp/modules/Client.js

```diff
@@ -1416,8 +1450,31 @@ Guacamole.Client = function(tunnel) {
         },
         
         "msg" : function(parameters) {
-            
-            if (guac_client.onmsg) guac_client.onmsg(parseInt(parameters[0]), parameters.slice(1));
+
+            var userID;
+            var username;
+
+            var msgid = parseInt(parameters[0]);
+            if (guac_client.onmsg)
+                guac_client.onmsg(msgid, parameters.slice(1));
+
+            switch (msgid) {
+
+                case Guacamole.Client.Message.USER_JOINED:
+                    userID = parameters[1];
+                    username = parameters[2];
+                    if (guac_client.onjoin)
+                        guac_client.onjoin(userID, username);
+                    break;
+
+                case Guacamole.Client.Message.USER_LEFT:
+                    userID = parameters[1];
+                    username = parameters[2];
+                    if (guac_client.onleave)
+                        guac_client.onleave(userID, username);
+                    break;
+
+            }
```

Comment:
The way this is implemented, this will fire _both_ `guac_client.onmsg()`, if it's defined, as well as `guac_client.onjoin()` or `guac_client.onleave()`, if those are defined. I just want to make sure that's the intended behavior and how you see that implementation working? I'm not sure if there are other instructions that are similarly implemented, or if this is a first of its kind?

Reply:
What about firing `onmsg` first, and then `onjoin`/`onleave` if the default behavior isn't prevented by the `onmsg` handler?

Reply:
I'd be fine with that - or adding a `default:` to the `switch()` statement?

I don't know that I have a strong preference one way or the other - mainly I just wanted to make sure that firing both the `onmsg()` handler and then the `onjoin()` or `onleave()` handler was the intentional and not any sort of an "oops, forgot to handle `return` the `onmsg()` call, etc.

Reply:
Ah, gotcha. Nope - definitely on purpose. I wanted to provide an optional, more convenient way of handling these messages that doesn't require as much low-level familiarity with the Guacamole protocol.

Reply:
I'll push a commit with the revised behavior momentarily - `onmsg` first, followed by `onjoin`/`onleave` if default behavior is not prevented.

Reply:
Commit pushed: bd913274152fcc12a12e26955bb3acdaa71e52e8